### ISSUE TITLE
Cannot reproduce result with given checkpoints

### ISSUE THREAD

**Issue by VuHuy-cse-9** [Ehzoahis/DEReD#6]
Hi.
With the camera setting in gen_nyuv2_defocus.py, I cannot reproduce the result in your paper using your checkpoints. The result is reall worse.
Here is my result, I use Orig checkpoints. The Noise checkpoints is also the same.
[evalNone.csv](https://github.com/Ehzoahis/DEReD/files/13255906/evalNone.csv)
Here is the depth prediction:
![reverse_darkmde-Page-4](https://github.com/Ehzoahis/DEReD/assets/57817188/ef196ac4-a36b-45cc-8389-79c335cd3e1d)
Here is the focal stack:
![reverse_darkmde-Page-4 (1)](https://github.com/Ehzoahis/DEReD/assets/57817188/3aca28b5-1405-4d40-9439-e29b2a89f4a1)
I also change the 'focus_dist' of NYV2 option in file data_config.py: 'focus_dist': [1, 1.5, 2.5, 4, 6]

**Comment by Ehzoahis** [comment]
The generated defocus map seems incorrect, have you checked the gaussian_psf code? You can check it by running the code in a Jupyter notebook and visualize the defocus map (output by the camera) and check if the generated defocus maps and defocus images make sense.

**Comment by VuHuy-cse-9** [comment]
I have visualized it herer: It's seem all defocus map similar.
![image](https://github.com/Ehzoahis/DEReD/assets/57817188/1dde480d-8929-48d3-88c0-a81650bf591f)

**Comment by VuHuy-cse-9** [comment]
I have visualized it herer: It's seem all defocus map similar.
![image](https://github.com/Ehzoahis/DEReD/assets/57817188/1dde480d-8929-48d3-88c0-a81650bf591f)

**Comment by Ehzoahis** [comment]
Is your NYUv2 depth normalized? Can you also visualize the colorbar?

**Comment by VuHuy-cse-9** [comment]
Here I visualize the colorbar:
![image](https://github.com/Ehzoahis/DEReD/assets/57817188/37f15b9e-b715-43f7-add2-11286f5150b3)

**Comment by Ehzoahis** [comment]
Can you also check the range of depth for NYUv2 you used?

**Comment by VuHuy-cse-9** [comment]
It turn out that I use depth with 16bit png, so I change the divisor from 1e4 into 256 in line 31, gen_nyuv2_defocus.py, and line 90, dataset.py. The depth range for this image is 7.0 - 14m. The focal stack is now more convincible. I also visualize the defocus map.
Here i my new result.
[evalNone.csv](https://github.com/Ehzoahis/DEReD/files/13258092/evalNone.csv)
Here is the focal stack:
![image](https://github.com/Ehzoahis/DEReD/assets/57817188/9f6a149c-3889-46ac-bdaa-9af38cc20612)
Here is the defocus map:
![image](https://github.com/Ehzoahis/DEReD/assets/57817188/4e9656e4-0af4-4d39-b6c8-3d09b8611501)
Here is the prediction of model:
![image](https://github.com/Ehzoahis/DEReD/assets/57817188/854d56ac-7b40-498f-aea9-1a9a8a621742)
I use "pred_dpt" variable to visualize depth map, the range of depth prediction is [min 0.10000000149011612 max 9.990567207336426]

**Comment by Ehzoahis** [comment]
As far as I remember, the depth map of NYUv2 is ranging from 0-10m. Can you also provide a colorbar of your predicted depth map?

**Comment by VuHuy-cse-9** [comment]
Here is the predicted depth map with colorbar: 
![image](https://github.com/Ehzoahis/DEReD/assets/57817188/325d76ec-ff07-45c2-bb3f-02dda69ba329)

**Comment by VuHuy-cse-9** [comment]
You are right! As i use nyuv2-toolkit from here: https://github.com/VainF/nyuv2-python-toolkit, they multiply depthmap with 1e3. So, I change the divisor from 256 to 1e3, the predict depth is more reliable.
Here is the predicted depth map:
![FrozenLake-Page-2](https://github.com/Ehzoahis/DEReD/assets/57817188/6e672925-538a-4623-9d78-04669183c18a)
However, the result is still different:
[evalNone.csv](https://github.com/Ehzoahis/DEReD/files/13259250/evalNone.csv)
The average result is: AbsRel 0.054 RMSE 0.359 delta1 0.947 delta2 0.982 delta3 0.996
I expect it should like table-2, Fully Self-Supervised Learning:
AbsRel 0.170 RMSE 0.325 delta1 0.950 delta2 0.979 delta3 0.987

**Comment by Ehzoahis** [comment]
Would you please check whether the gt depth range is 0-10m in this case? Your overall depth still seems to be deeper than my results if you check the supplementary of my paper.

**Comment by VuHuy-cse-9** [comment]
Sorry for late response, from your suggestion, I tried to figure out the bug but I couldn't. Our gt depth range is 0 - 10. Here are some examples (I use Orig model to estimate):
![FrozenLake-Page-2 (1)](https://github.com/Ehzoahis/DEReD/assets/57817188/d33876fa-98c4-4348-a19f-a578d218c317)
Like in your supplementary, I also visualize the focus map with colorbar, it's a bit different from your supplementary.
![FrozenLake-Page-3](https://github.com/Ehzoahis/DEReD/assets/57817188/dd6d690f-16e7-456e-88da-58e262979e94)

**Comment by Ehzoahis** [comment]
What is the resolution of your data? Did you downsample the input as I did? Also, I noticed that our RGB data is not exactly the same in color. I would imagine the slight model performance variance is due to some difference between our data. I would suggest you perform a simple finetuning if you really want to achieve the same performance as reported.

**Comment by VuHuy-cse-9** [comment]
The image size is 320x240. As I see in eval_model, the depth prediction is upsampled to 640x320. What bothers me is that the Abs rel - the most important metric for me - dropped a half. 
I'm currently fine-tuning & reproducing your work. This is an exciting approach, and many potential works can improve the performance. Thanks for your work & advice.

**Comment by VuHuy-cse-9** [comment]
Hi, I cannot reproduce result after fine-tuning or training.
For fine-tuning, I trained for more than 200 epochs, batch-size = 16, other setting remain the same:
`python train.py --data_path /workspace/datasets/nyuv2/NYUv2\
                        --dataset NYUv2 --recon_all \
                        -N NYUv2_fine_tuning --use_cuda -E 1200 --BS 16 --save_checkpoint --save_best --save_last \
                        --sm_loss_beta 2.5 --verbose --recon_loss_lambda 1e3 --aif_blur_loss_lambda 10 \
                        --blur_loss_lambda 1e1 --sm_loss_lambda 1e1 --log --vis --continue_from "/workspace/log/DEReD/DAIF_NYUv2/Orig/model/best-model.pth"`

From paper: abs_rel 0.170 rmse 0.325 a1 0.950 a2 0.979 a3 0.987
Orig-reeval: abs_rel 0.054 rmse 0.359 a1 0.947 a2 0.982 a3 0.996
Orig-Fine-tun: abs_rel 0.054 rmse 0.357 a1 0.948 a2 0.983 a3 0.996

I also trained from stratch, with more than 500 epochs, batch-size = 16, other setting remain the same:

DERED	abs_rel 0.955 sq_rel 2.544 rmse 2.831 rmse log 1.400 a1 0.000 a2 0.000	a3 0.000

From wandb, the loss didn't converge: [Wandb](https://wandb.ai/fpt-aia-aisa-face-generator/SS-DFD-Formal/reports/DERED-Fine-tuning--Vmlldzo1ODkzMTgy)

Here is the depth prediction from model: 
![DERED](https://github.com/Ehzoahis/DEReD/assets/57817188/f5ce17ab-28fc-4269-bba3-fd305089527a)

I use the same software verion specified in requirements.txt.

**Comment by Ehzoahis** [comment]
The performance variance may also be due to the hardware. You may also adjust the hparam when retraining. Also, your result is better than the reported results on 3 out of 5 metrics. I would think it is already a successful finetuned result.

**Comment by VuHuy-cse-9** [comment]
How about the re-training result, I think it didn't converge.

**Comment by Ehzoahis** [comment]
I cannot see your wandb, but you may want to adjust the loss weights according to the training curves.

**Comment by VuHuy-cse-9** [comment]
Sorry, I have opened the wandb link. I would decrease the learning rate. The depth seems really worse, even though the network be trained for 500 epochs.

**Comment by Ehzoahis** [comment]
Still, I have no access to your wandb. Also, as I suggested, you may need to adjust the hyperparameter before training from scratch according to your current training curve.

**Comment by VuHuy-cse-9** [comment]
Okay, thanks you for your advice. 
Here is the new wandb link: [wandb-share](https://api.wandb.ai/links/fpt-aia-aisa-face-generator/czl9mvmi)

**Comment by VuHuy-cse-9** [comment]
The above link is fine-tuning version. Here is the training from stratch wandb: [wandb-from-stratch](https://api.wandb.ai/links/fpt-aia-aisa-face-generator/ptn09bkr)

**Comment by Ehzoahis** [comment]
It seems you didn't visualize the defocus_recon loss.

**Comment by VuHuy-cse-9** [comment]
I'm a newble to wandb. Here is the log for all loss: [Wandb-reproduce-all](https://api.wandb.ai/links/fpt-aia-aisa-face-generator/hjjas4q5)

**Comment by Ehzoahis** [comment]
You probably need to increase your lambda for recon loss. The performance difference may be due to the difference between hardware.

**Comment by VuHuy-cse-9** [comment]
It converges after I change  recon_loss_lambda = 1e4 and aif_recon_loss_lambda=1e3.
Thanks for your work and advice :).

**Comment by no-Seaweed** [comment]
> What is the resolution of your data? Did you downsample the input as I did? Also, I noticed that our RGB data is not exactly the same in color. I would imagine the slight model performance variance is due to some difference between our data. I would suggest you perform a simple finetuning if you really want to achieve the same performance as reported.

I think the difference is from RGB2BGR and horizontal flip. Did you perform those operations when you processing NYUv2?
